src/server/pfs/pfs.go

```diff
@@ -429,7 +429,7 @@ func IsDropWithChildrenErr(err error) bool {
 func ValidateSQLDatabaseEgress(sql *pfs.SQLDatabaseEgress) error {
 	if sql != nil {
 		if secret := sql.GetSecret(); secret != nil {
-			if secret.K8SSecret != "" && secret.Key != "" {
+			if secret.Name != "" {
```

Comment:
If `Key` is empty, what are we going to use as the name of the key?  I guess the empty string is a string, but if you try to create a secret with a key that is the empty string, you get:

```
# secrets "foo" was not valid:
# * data[]: Invalid value: "": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')
```

Reply:
Oh oops didn't mean to remove the check on the key.